saleor/checkout/base_calculations.py

```diff
@@ -0,0 +1,63 @@
+"""Contain functions which are base for calculating checkout properties.
+
+It's recommended to use functions from calculations.py module to take in account plugin
+manager.
+"""
+
+from typing import TYPE_CHECKING
+
+from prices import TaxedMoney
+
+from ..core.taxes import quantize_price, zero_taxed_money
+from ..extensions.manager import get_extensions_manager
+
+if TYPE_CHECKING:
+    from .models import Checkout, CheckoutLine
+    from ..discount.types import DiscountsListType
+
+
+def get_base_checkout_shipping_price(
+    checkout: "Checkout", discounts: "DiscountsListType" = None
+) -> TaxedMoney:
+    """Return checkout shipping price."""
+    # FIXME: should it take discounts in account?
+    if not checkout.shipping_method or not checkout.is_shipping_required():
+        return zero_taxed_money(checkout.currency)
+
+    shipping_price = checkout.shipping_method.get_total()
+    return quantize_price(
+        TaxedMoney(net=shipping_price, gross=shipping_price), shipping_price.currency
+    )
+
+
+def get_base_checkout_subtotal(
```

Comment:
I am not sure of these methods. Technically, they just trigger an `extension_manager`. They don't do any `base_` logic. Maybe it should be directly done in the `Manager` methods?
We import manager here and inside manager file we import `base_calulcation` file, which means that we could wrongly place some methods.

Reply:
They are `base` not because they are not using mgr, but because that's most basic algorithm for calculating those values - not taking manager in the account will produce wrong results. Since we introduced plugins there is no such thing as 'subtotal without plugins'.

The idea behind file `calculations.py` is having basic checkout utils inside the checkout module and using those, not having to go through tens of manager methods. 

>

Reply:
I like the fact that we move calculations from manager. I think manager's reponsibility should be to call the right function instead of containing any business logic. I'm not sure about introducing two files though - `base_calculations` and `calculations`. I'm worried that it may be confusing in the future what's the difference between them and where should I add my new calculation. Couldn't the "base calculations" sit inside `Checkout` model?

Reply:
Probably if you want new calculation, then you should write plugin :)